src/Type/IntersectionType.php

```diff
@@ -348,6 +348,22 @@ private function describeItself(VerbosityLevel $level, bool $skipAccessoryTypes)
 				continue;
 			}
 
+			if (
+				substr($typeDescription, 0, strlen('array{')) === 'array{'
+					&& in_array('array', $skipTypeNames, true)
+			) {
+				$isList = false;
+				foreach ($typesToDescribe as $j => $typeToDescribe) {
+					if ($typeToDescribe instanceof AccessoryArrayListType) {
+						$isList = true;
+						unset($typesToDescribe[$j]);
+						break;
+					}
+				}
+				$describedTypes[$i] = 'list{' . substr($typeDescription, strlen('array{'));
+				continue;
+			}
```

Comment:
> 3. In `IntersectionType::describeItself` there's a section that makes sure that `array<...>&list` is described as `list<...>`. Please write similar logic there so that `array{...}&list` is described as `list{...}`.

> `array{0: 'a', 1: 'b'}` is always a list so that's why the `&list` part is omitted and that's why it's not an intersection type. So we probably don't need to worry about this at all. The point of `list{...}` should be to reject types like `list{foo: 1}`.


As you say `array{0: 'a', 1: 'b'}&list` is always `ConstantArrayType` not `IntersectionType`. Changing `IntersectionType::describeItself()` for print `list{}` seems to have no effect.

> Reset everything in src/, leave just the tests.

Maybe I need to reapply the reverted fa7bcee09da8565bfe9f5a557c2894922f74faf1 and change  `ConstantArrayType`.

Reply:
Do not overthink this :) It's fine to describe `list{1}` in PHPDoc as `array{1}` with dumpType.

Reply:
I think you can delete this change without anything failing.